samza-core/src/main/scala/org/apache/samza/container/SamzaContainer.scala

```diff
@@ -904,6 +907,17 @@ class SamzaContainer(
     })
   }
 
+  def startDiagnostics {
+    if (containerContext.config.getDiagnosticsEnabled) {
+      val rootLogger = Logger.getRootLogger
+
+      if (rootLogger.getAppender(classOf[DiagnosticsAppender].getName) == null) {
```

Comment:
It seems that in log4j2, the API to get appender dynamically also changed: https://logging.apache.org/log4j/2.x/log4j-core/apidocs/org/apache/logging/log4j/core/config/Configuration.html#getAppender(java.lang.String). Please check w/ @PawasChhokra to see what's the best way to make sure the code here is log4j version agnostic.